polkadot/runtime/parachains/src/assigner_coretime/mod.rs

```diff
@@ -387,67 +380,56 @@ impl<T: Config> Pallet<T> {
 
 	/// Append another assignment for a core.
 	///
-	/// Important only appending is allowed. Meaning, all already existing assignments must have a
-	/// begin smaller than the one passed here. This restriction exists, because it makes the
-	/// insertion O(1) and the author could not think of a reason, why this restriction should be
-	/// causing any problems. Inserting arbitrarily causes a `DispatchError::DisallowedInsert`
-	/// error. This restriction could easily be lifted if need be and in fact an implementation is
-	/// available
+	/// Important only appending is allowed or insertion into the last item is possible. Meaning,
+	/// all already existing assignments must have a begin smaller or equal than the one passed
+	/// here. This restriction exists, because it makes the insertion O(1) and the author could not
+	/// think of a reason, why this restriction should be causing any problems. Inserting
+	/// arbitrarily causes a `DispatchError::DisallowedInsert` error. This restriction could easily
+	/// be lifted if need be and in fact an implementation is available
 	/// [here](https://github.com/paritytech/polkadot-sdk/pull/1694/commits/c0c23b01fd2830910cde92c11960dad12cdff398#diff-0c85a46e448de79a5452395829986ee8747e17a857c27ab624304987d2dde8baR386).
 	/// The problem is that insertion complexity then depends on the size of the existing queue,
 	/// which makes determining weights hard and could lead to issues like overweight blocks (at
```

Comment:
Implementation details should not be discussed in the documentation of the function. What does the user gets from this? 

If you really want to keep this, it should be a code comment, but not sure there is such a big value in this.